cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/polling/task/ClouderaManagerParcelDeletedListenerTask.java

```diff
@@ -0,0 +1,91 @@
+package com.sequenceiq.cloudbreak.cm.polling.task;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.cloudera.api.swagger.CommandsResourceApi;
+import com.cloudera.api.swagger.ParcelsResourceApi;
+import com.cloudera.api.swagger.client.ApiClient;
+import com.cloudera.api.swagger.client.ApiException;
+import com.cloudera.api.swagger.model.ApiParcel;
+import com.cloudera.api.swagger.model.ApiParcelList;
+import com.sequenceiq.cloudbreak.cm.ClouderaManagerOperationFailedException;
+import com.sequenceiq.cloudbreak.cm.client.ClouderaManagerApiPojoFactory;
+import com.sequenceiq.cloudbreak.cm.polling.ClouderaManagerPollerObject;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.structuredevent.event.CloudbreakEventService;
+
+public class ClouderaManagerParcelDeletedListenerTask extends AbstractClouderaManagerCommandCheckerTask<ClouderaManagerPollerObject> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClouderaManagerParcelDeletedListenerTask.class);
+
+    private Map<String, String> parcelVersion;
+
+    public ClouderaManagerParcelDeletedListenerTask(ClouderaManagerApiPojoFactory clouderaManagerApiPojoFactory,
+            CloudbreakEventService cloudbreakEventService, Map<String, String> parcelVersions) {
+        super(clouderaManagerApiPojoFactory, cloudbreakEventService);
+        this.parcelVersion = parcelVersions;
+    }
+
+    @Override
+    protected boolean doStatusCheck(ClouderaManagerPollerObject pollerObject, CommandsResourceApi commandsResourceApi) throws ApiException {
+        ApiClient apiClient = pollerObject.getApiClient();
+        Stack stack = pollerObject.getStack();
+        ApiParcelList parcels = getClouderaManagerParcels(apiClient, stack.getName());
+        List<ApiParcel> existedParcels = getExistedParcels(parcelVersion, parcels);
+        if (existedParcels.isEmpty()) {
+            LOGGER.debug("Parcels are deleted succesfully.");
+            return true;
+        } else {
+            LOGGER.debug("Some parcels are not yet deleted: [{}].", getJoinedParcelStages(existedParcels));
```

Comment:
The second upgrade stuck for me at checking the deleted parcels
```
Some parcels are not yet deleted: [(CDH 7.2.1-1.cdh7.2.1.p0.4763198 : AVAILABLE_REMOTELY)].
```
We're checking if it's empty, but for me, it shows AVAILABLE_REMOTELY on the UI. We can consider this one as deleted as well.